docs/source/libraries/xgboost.rst

```diff
@@ -37,8 +38,13 @@ of an ensemble. Each node of the tree has an output score, and
 contribution of a feature on the decision path is how much the score changes
 from parent to child.
 
+.. note::
+    When explaining Booster_ predictions,
+    do not pass an ``xgboost.DMatrix`` object as ``doc``, pass a numpy array
+    or a sparse matrix instead (or have ``vec`` return them).
```

Comment:
There is a deeper reason, I don't think we'll be able to support it. The reason is that ``DMatrix`` is completely opaque to the python code - you can not convert it to numpy array without dumping to disk (and using some external code to parse it), or read values from it - but we need to be able to read individual values to show the feature values. Due to this, I think it's not a huge drawback - probably people that use DMatrix create it just for training or batch prediction, but also have numpy data that is used for creating DMatrix.